src/src/DataStructs/ControllerSettingsStruct.cpp

```diff
@@ -288,3 +313,58 @@ void ControllerSettingsStruct::useLocalSystemTime(bool value)
 {
   bitWrite(VariousFlags, 11, value);
 }
+
+TLS_types ControllerSettingsStruct::TLStype() const
+{
+  // Store it in bits 12, 13, 14, 15
+  const TLS_types tls_type = static_cast<TLS_types>((VariousFlags >> 12) & 0xF);
+  return tls_type;
+}
+
+void  ControllerSettingsStruct::TLStype(TLS_types tls_type)
+{
+  const uint32_t mask = ~(0xF);
+  VariousFlags &= mask; // Clear the bits
+  const uint32_t tls_type_val = static_cast<uint32_t>(tls_type) << 12;
+  VariousFlags |= tls_type_val;
+}
```

Comment:
This can be replaced by `set4BitToUL(VariousFlags, 12, static_cast<uint8_t>(tls_type))` helper function from `Misc.h/cpp`.